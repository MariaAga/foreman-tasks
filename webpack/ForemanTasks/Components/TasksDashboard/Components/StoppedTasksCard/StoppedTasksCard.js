import React from 'react';
import PropTypes from 'prop-types';
import { Card } from 'patternfly-react';
import classNames from 'classnames';
import './StoppedTasksCard.scss';

class StoppedTasksCard extends React.Component {
  getTableItemClassName = (item, type) => {
    const { focusedOn } = this.props;
    return focusedOn.normal ||
      focusedOn.total ||
      (focusedOn[item] && focusedOn[item][type])
      ? `${type}-data`
      : 'not-focused';
  };
  shouleBeSelected = focusedOn =>
    focusedOn &&
    (focusedOn.total ||
      ((focusedOn.warning &&
        (focusedOn.warning.total || focusedOn.warning.last)) ||
        (focusedOn.success &&
          (focusedOn.success.total || focusedOn.success.last))) ||
      (focusedOn.error && (focusedOn.error.total || focusedOn.error.last)));
  render() {
    const {
      className,
      error,
      warning,
      success,
      timePeriod,
      focusedOn,
      onTitleClick,
    } = this.props;
    const data = { error, warning, success };

    return (
      <Card
        className={classNames(
          'tasks-donut-card',
          'stopped-tasks-card',
          className,
          {
            'selected-tasks-card': this.shouleBeSelected(focusedOn),
          }
        )}
      >
        <Card.Title onClick={onTitleClick}>Stopped</Card.Title>
        <Card.Body>
          <table className="table table-titles">
            <tbody>
              <tr>
                <td />
                <td>Total</td>
                <td>{timePeriod}</td>
              </tr>
            </tbody>
          </table>
          <table className="table table-bordered table-striped">
            <tbody>
              {Object.keys(data).map((item, index) => (
                <tr className={`${item}-row`} key={index}>
                  <td>{item[0].toUpperCase() + item.slice(1)}</td>
                  <td
                    className={this.getTableItemClassName(item, 'total')}
                    onClick={data[item].total.onClick}
                  >
                    {data[item].total.value}
                  </td>
                  <td
                    className={this.getTableItemClassName(item, 'last')}
                    onClick={data[item].last.onClick}
                  >
                    {data[item].last.value}
                  </td>
                </tr>
              ))}
            </tbody>
          </table>
        </Card.Body>
      </Card>
    );
  }
}

StoppedTasksCard.propTypes = {
  className: PropTypes.string,
  focusedOn: PropTypes.shape({
    normal: PropTypes.bool,
    error: PropTypes.shape({
      total: PropTypes.bool,
      last: PropTypes.bool,
    }),
    warning: PropTypes.shape({
      total: PropTypes.bool,
      last: PropTypes.bool,
    }),
    success: PropTypes.shape({
      total: PropTypes.bool,
      last: PropTypes.bool,
    }),
  }),
  onTitleClick: PropTypes.func,
  error: PropTypes.shape({
    total: { value: PropTypes.number, onClick: PropTypes.func },
    last: { value: PropTypes.number, onClick: PropTypes.func },
  }),
  warning: PropTypes.shape({
    total: { value: PropTypes.number, onClick: PropTypes.func },
    last: { value: PropTypes.number, onClick: PropTypes.func },
  }),
  success: PropTypes.shape({
    total: { value: PropTypes.number, onClick: PropTypes.func },
    last: { value: PropTypes.number, onClick: PropTypes.func },
  }),
  timePeriod: PropTypes.string,
};

StoppedTasksCard.defaultProps = {
  className: '',
  focusedOn: {},
  onTitleClick: () => null,
  error: { total: 0, last: 0 },
  warning: { total: 0, last: 0 },
  success: { total: 0, last: 0 },
  timePeriod: '24h',
};

export default StoppedTasksCard;
