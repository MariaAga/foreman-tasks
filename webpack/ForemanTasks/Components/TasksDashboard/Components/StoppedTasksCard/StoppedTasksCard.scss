.stopped-tasks-card {
  table.stopped-table {
    width: 100%;
    border: none;
    table-layout: fixed;

    &.not-focused {
      opacity: 0.3;
    }

    thead {
      background-color: transparent !important;
      background-image: none !important;
      tr {
        th {
          font-weight: bold !important;
          border: none !important;
        }
      }
    }

    td, th {
      text-align: center;
      padding-left: 5px;
      width: 33%;

      &:first-child {
        text-align: left;
      }
    }

    td {
      &:hover {
        cursor: pointer;
        border: 1px solid #d1d1d1 !important;
        background-color: #fff !important;

        &:first-child {
          cursor: auto;
        }
      }

      &.active {
        font-weight: bold;
      }
    }

    .error-row {
      .total-col {
        background-color: #fcc5ca;
      }
      .last-col {
        background-color: #f78c96;
      }
    }
    .warning-row {
      .total-col {
        background-color: #fee4cc;
      }
      .last-col {
        background-color: #fdc899;
      }
    }
    .success-row {
      .total-col {
        background-color: #d1f0d6;
      }
      .last-col {
        background-color: #a4e0ac;
      }
    }
  }
}
