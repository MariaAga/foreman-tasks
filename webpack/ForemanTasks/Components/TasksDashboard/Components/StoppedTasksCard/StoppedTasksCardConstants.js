export const STOPPED_TASK_CARD_FOCUSED_ON_OPTIONS = {
  NORMAL: 'normal', // normal-mode
  STOPPED_TOTAL: 'stopped-total', // total-mode,
  ERROR_TOTAL: 'stopped-total-error', // error total-mode
  ERROR_LAST: 'stopped-last-error', // error last X mode
  WARNING_TOTAL: 'stopped-total-warning', // warning total-mode
  WARNING_LAST: 'stopped-last-warning', // warning last X mode
  SUCCESS_TOTAL: 'stopped-total-success', // success total-mode
  SUCCESS_LAST: 'stopped-last-success', // success last X mode
  NONE: 'none', // unfocus-mode: another card is selected
};

export const STOPPED_TASK_CARD_FOCUSED_ON_OPTIONS_ARRAY = Object.values(
  STOPPED_TASK_CARD_FOCUSED_ON_OPTIONS
);
