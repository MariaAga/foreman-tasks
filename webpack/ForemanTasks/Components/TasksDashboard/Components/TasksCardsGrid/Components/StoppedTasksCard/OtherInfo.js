import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import { Icon, Button, OverlayTrigger, Tooltip } from 'patternfly-react';
import { translate as __ } from 'foremanReact/common/I18n';
import {
  TASKS_DASHBOARD_AVAILABLE_QUERY_STATES,
  TASKS_DASHBOARD_AVAILABLE_QUERY_RESULTS,
} from '../../../../TasksDashboardConstants';
import { queryPropType } from '../../../../TasksDashboardPropTypes';

const tooltip = (
  <Tooltip id="stopped-tooltip">
    {__('Other includes all stopped tasks that are cancelled or pending')}
  </Tooltip>
);

export const OtherInfo = ({ updateQuery, otherCount, query }) => {
  const { OTHER } = TASKS_DASHBOARD_AVAILABLE_QUERY_RESULTS;
  const { STOPPED } = TASKS_DASHBOARD_AVAILABLE_QUERY_STATES;
  const active = query.state === STOPPED && query.result === OTHER;
  return (
    <span className={classNames(active && 'other-active')}>
      <OverlayTrigger
        overlay={tooltip}
        trigger={['hover', 'focus']}
        placement="bottom"
      >
        <span>
          <Icon type="pf" name="info" />
          <span>{__('Other:')} </span>
        </span>
      </OverlayTrigger>
      <Button
        bsStyle="link"
        onClick={() =>
          updateQuery({
            state: STOPPED,
            result: OTHER,
          })
        }
      >
        {otherCount}
      </Button>
    </span>
  );
};
OtherInfo.propTypes = {
  updateQuery: PropTypes.func.isRequired,
  otherCount: PropTypes.number.isRequired,
  query: queryPropType.isRequired,
};
