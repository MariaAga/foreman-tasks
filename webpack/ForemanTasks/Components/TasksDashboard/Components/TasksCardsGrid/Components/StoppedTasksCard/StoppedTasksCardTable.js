import React from 'react';
import PropTypes from 'prop-types';
import { capitalize } from 'lodash';
import classNames from 'classnames';
import { Icon, Button } from 'patternfly-react';
import { translate as __ } from 'foremanReact/common/I18n';
import {
  TASKS_DASHBOARD_AVAILABLE_QUERY_STATES,
  TASKS_DASHBOARD_AVAILABLE_QUERY_MODES,
} from '../../../../TasksDashboardConstants';
import { getQueryValueText } from '../../../../TasksDashboardHelper';
import {
  timePropType,
  queryPropType,
} from '../../../../TasksDashboardPropTypes';

const resultIcons = {
  error: <Icon type="pf" name="error-circle-o" />,
  warning: <Icon type="pf" name="warning-triangle-o" />,
  success: <Icon type="pf" name="ok" />,
};

const StoppedTableCells = (data, query, time, updateQuery) =>
  Object.entries(data).map(([result, { total, last }]) => {
    const { STOPPED } = TASKS_DASHBOARD_AVAILABLE_QUERY_STATES;
    const { LAST } = TASKS_DASHBOARD_AVAILABLE_QUERY_MODES;
    const { state, mode } = query;

    const active = state === STOPPED && query.result === result;
    const activeTotal = active && mode !== LAST;
    const activeLast = active && mode === LAST && query.time === time;
    const notFocusedTotal =
      state && !(state === STOPPED && !query.result) && !activeTotal;
    const notFocusedLast =
      state && !(state === STOPPED && !query.result) && !activeLast;

    return (
      <tr key={result}>
        <td>
          {resultIcons[result]}
          {capitalize(result)}
        </td>
        <td
          className={classNames('data-col', {
            active: active && mode !== LAST,
            'not-focused': notFocusedTotal,
          })}
          onClick={() => updateQuery({ state: STOPPED, result })}
        >
          <Button bsStyle="link">{total}</Button>
        </td>
        <td
          className={classNames('data-col', {
            active: activeLast,
            'not-focused': notFocusedLast,
          })}
          onClick={() =>
            updateQuery({
              state: STOPPED,
              result,
              mode: LAST,
              time,
            })
          }
        >
          <Button bsStyle="link">{last}</Button>
        </td>
      </tr>
    );
  });

export const StoppedTable = ({ data, query, time, updateQuery }) => (
  <table className="table table-bordered table-striped stopped-table">
    <thead>
      <tr>
        <th />
        <th>{__('Total')}</th>
        <th>{getQueryValueText(time)}</th>
      </tr>
    </thead>
    <tbody>{StoppedTableCells(data, query, time, updateQuery)}</tbody>
  </table>
);

StoppedTable.propTypes = {
  data: PropTypes.object.isRequired,
  query: queryPropType.isRequired,
  time: timePropType.isRequired,
  updateQuery: PropTypes.func.isRequired,
};
