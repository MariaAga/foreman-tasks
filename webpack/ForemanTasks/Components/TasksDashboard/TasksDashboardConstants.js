import { translate as __ } from 'foremanReact/common/I18n';

export const TASKS_SUMMARY_REQUEST = 'TASKS_SUMMARY_REQUEST';
export const TASKS_SUMMARY_SUCCESS = 'TASKS_SUMMARY_SUCCESS';
export const TASKS_SUMMARY_FAILURE = 'TASKS_SUMMARY_FAILURE';

export const TASKS_SUMMARY_ZERO = {
  running: {
    recent: 0,
    total: 0,
  },
  paused: {
    recent: 0,
    total: 0,
  },
  stopped: {
    by_result: {
      error: {
        total: 0,
        recent: 0,
      },
      warning: {
        total: 0,
        recent: 0,
      },
      success: {
        total: 0,
        recent: 0,
      },
    },
  },
  scheduled: {
    total: 0,
  },
};

export const TASKS_DASHBOARD_AVAILABLE_QUERY_STATES = {
  RUNNING: 'running',
  PAUSED: 'paused',
  STOPPED: 'stopped',
  SCHEDULED: 'scheduled',
};

export const TASKS_DASHBOARD_AVAILABLE_QUERY_RESULTS = {
  ERROR: 'error',
  WARNING: 'warning',
  SUCCESS: 'success',
};

export const TASKS_DASHBOARD_AVAILABLE_QUERY_MODES = {
  LAST: 'last',
  OLDER: 'older',
};

export const TASKS_DASHBOARD_AVAILABLE_TIMES = {
  H24: 'H24',
  H12: 'H12',
  WEEK: 'WEEK',
};

export const TASKS_DASHBOARD_AVAILABLE_TIMES_TEXT = {
  [TASKS_DASHBOARD_AVAILABLE_TIMES.H24]: __('24h'),
  [TASKS_DASHBOARD_AVAILABLE_TIMES.H12]: __('12h'),
  [TASKS_DASHBOARD_AVAILABLE_TIMES.WEEK]: __('week'),
};

export const STATUS = {
  PENDING: 'PENDING',
  RESOLVED: 'RESOLVED',
  ERROR: 'ERROR',
};
