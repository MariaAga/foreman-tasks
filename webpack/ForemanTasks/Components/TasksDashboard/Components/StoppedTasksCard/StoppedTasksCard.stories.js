import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, number, select } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';
import { withCardsDecorator } from '../../../../../stories/decorators';

import {
  TASKS_DASHBOARD_AVAILABLE_TIMES,
  TASKS_DASHBOARD_AVAILABLE_QUERY_STATES,
  TASKS_DASHBOARD_AVAILABLE_QUERY_RESULTS,
  TASKS_DASHBOARD_AVAILABLE_QUERY_MODES,
} from '../../TasksDashboardConstants';
import StoppedTasksCard from './StoppedTasksCard';

storiesOf('TasksDashboard', module)
  .addDecorator(withKnobs)
  .addDecorator(withCardsDecorator)
  .add('StoppedTasksCard', () => {
    const selectTime = select(
      'time',
      TASKS_DASHBOARD_AVAILABLE_TIMES,
      StoppedTasksCard.defaultProps.time
    );
    const selectState = select(
      'query.state',
      { ...TASKS_DASHBOARD_AVAILABLE_QUERY_STATES, NONE: 'none' },
      'none'
    );
    const selectResult = select(
      'query.result',
      { ...TASKS_DASHBOARD_AVAILABLE_QUERY_RESULTS, NONE: 'none' },
      'none'
    );
    const selectMode = select(
      'query.mode',
      { ...TASKS_DASHBOARD_AVAILABLE_QUERY_MODES, NONE: 'none' },
      'none'
    );
    return (
      <StoppedTasksCard
        data={{
          error: {
            total: number('errorTotal', 8),
            last: number('errorLast', 1),
          },
          warning: {
            total: number('warningTotal', 20),
            last: number('warningLast', 2),
          },
          success: {
            total: number('successTotal', 25),
            last: number('successLast', 3),
          },
        }}
        time={selectTime}
        query={{
          state: selectState,
          result: selectResult,
          mode: selectMode,
          time: selectTime,
        }}
        updateQuery={action('updateQuery')}
      />
    );
  });
