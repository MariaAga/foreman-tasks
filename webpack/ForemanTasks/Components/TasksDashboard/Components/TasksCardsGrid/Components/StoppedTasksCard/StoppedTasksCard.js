import React from 'react';
import PropTypes from 'prop-types';
import { Card } from 'patternfly-react';
import classNames from 'classnames';
import { capitalize } from 'lodash';
import { translate as __ } from 'foremanReact/common/I18n';

import {
  timePropType,
  queryPropType,
} from '../../../../TasksDashboardPropTypes';
import {
  TASKS_DASHBOARD_AVAILABLE_TIMES,
  TASKS_DASHBOARD_AVAILABLE_QUERY_STATES,
  TASKS_DASHBOARD_AVAILABLE_QUERY_MODES,
} from '../../../../TasksDashboardConstants';
import { getTimeText } from '../../../../TasksDashboardHelper';
import './StoppedTasksCard.scss';

const StoppedTasksCard = ({
  data,
  query,
  time,
  className,
  updateQuery,
  ...props
}) => {
  const { STOPPED } = TASKS_DASHBOARD_AVAILABLE_QUERY_STATES;
  const { LAST } = TASKS_DASHBOARD_AVAILABLE_QUERY_MODES;

  return (
    <Card
      className={classNames(
        'tasks-donut-card',
        'stopped-tasks-card',
        className,
        {
          'selected-tasks-card': query.state === STOPPED,
        }
      )}
      {...props}
    >
      <Card.Title onClick={() => updateQuery({ state: STOPPED })}>
        {__('Stopped')}
      </Card.Title>
      <Card.Body>
        <table className="table table-bordered table-striped stopped-table">
          <thead>
            <tr>
              <th />
              <th>{__('Total')}</th>
              <th>{getTimeText(time)}</th>
            </tr>
          </thead>
          <tbody>
            {Object.entries(data).map(([result, { total, last }]) => {
              const active = query.state === STOPPED && query.result === result;
              const activeTotal = active && query.mode !== LAST;
              const activeLast =
                active && query.mode === LAST && query.time === time;
              return (
                <tr className={`${result}-row`} key={result}>
                  <td>{capitalize(result)}</td>
                  <td
                    className={classNames('total-col', {
                      active: activeTotal,
                      'not-focused':
                        query.state &&
                        !(query.state === STOPPED && !query.result) &&
                        !activeTotal,
                    })}
                    onClick={() => updateQuery({ state: STOPPED, result })}
                  >
                    {total}
                  </td>
                  <td
                    className={classNames('last-col', {
                      active: activeLast,
                      'not-focused':
                        query.state &&
                        !(query.state === STOPPED && !query.result) &&
                        !activeLast,
                    })}
                    onClick={() =>
                      updateQuery({
                        state: STOPPED,
                        result,
                        mode: LAST,
                        time,
                      })
                    }
                  >
                    {last}
                  </td>
                </tr>
              );
            })}
          </tbody>
        </table>
      </Card.Body>
    </Card>
  );
};

const resultPropType = PropTypes.shape({
  total: PropTypes.number.isRequired,
  last: PropTypes.number.isRequired,
});

StoppedTasksCard.propTypes = {
  data: PropTypes.shape({
    error: resultPropType.isRequired,
    warning: resultPropType.isRequired,
    success: resultPropType.isRequired,
  }),
  time: timePropType,
  query: queryPropType,
  className: PropTypes.string,
  updateQuery: PropTypes.func,
};

StoppedTasksCard.defaultProps = {
  data: {
    error: { total: 0, last: 0 },
    warning: { total: 0, last: 0 },
    success: { total: 0, last: 0 },
  },
  time: TASKS_DASHBOARD_AVAILABLE_TIMES.H24,
  query: {},
  className: '',
  updateQuery: () => null,
};

export default StoppedTasksCard;
