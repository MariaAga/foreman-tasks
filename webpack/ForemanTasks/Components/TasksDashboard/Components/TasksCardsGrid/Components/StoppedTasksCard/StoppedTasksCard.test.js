import { testComponentSnapshotsWithFixtures } from '@theforeman/test';
import { TASKS_DASHBOARD_AVAILABLE_QUERY_STATES } from '../../../../TasksDashboardConstants';
import StoppedTasksCard from './StoppedTasksCard';

const { STOPPED } = TASKS_DASHBOARD_AVAILABLE_QUERY_STATES;

const fixtures = {
  'render with minimal props': {},
  'render selected': {
    query: { state: STOPPED },
  },
};

describe('StoppedTasksCard', () =>
  testComponentSnapshotsWithFixtures(StoppedTasksCard, fixtures));
