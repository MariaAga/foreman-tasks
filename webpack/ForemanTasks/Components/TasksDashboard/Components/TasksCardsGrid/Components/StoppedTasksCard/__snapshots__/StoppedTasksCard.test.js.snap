// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoppedTasksCard render error-last selected 1`] = `
<Card
  accented={false}
  aggregated={false}
  aggregatedMini={false}
  cardRef={null}
  className="tasks-donut-card stopped-tasks-card selected-tasks-card"
  matchHeight={false}
>
  <CardTitle
    className=""
    onClick={[Function]}
  >
    Stopped
  </CardTitle>
  <CardBody
    className=""
  >
    <table
      className="table table-bordered table-striped stopped-table"
    >
      <thead>
        <tr>
          <th />
          <th>
            Total
          </th>
          <th>
            week
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          className="error-row"
          key="error"
        >
          <td>
            Error
          </td>
          <td
            className="total-col"
            onClick={[Function]}
          >
            0
          </td>
          <td
            className="last-col active"
            onClick={[Function]}
          >
            0
          </td>
        </tr>
        <tr
          className="warning-row"
          key="warning"
        >
          <td>
            Warning
          </td>
          <td
            className="total-col"
            onClick={[Function]}
          >
            0
          </td>
          <td
            className="last-col"
            onClick={[Function]}
          >
            0
          </td>
        </tr>
        <tr
          className="success-row"
          key="success"
        >
          <td>
            Success
          </td>
          <td
            className="total-col"
            onClick={[Function]}
          >
            0
          </td>
          <td
            className="last-col"
            onClick={[Function]}
          >
            0
          </td>
        </tr>
      </tbody>
    </table>
  </CardBody>
</Card>
`;

exports[`StoppedTasksCard render error-total selected 1`] = `
<Card
  accented={false}
  aggregated={false}
  aggregatedMini={false}
  cardRef={null}
  className="tasks-donut-card stopped-tasks-card selected-tasks-card"
  matchHeight={false}
>
  <CardTitle
    className=""
    onClick={[Function]}
  >
    Stopped
  </CardTitle>
  <CardBody
    className=""
  >
    <table
      className="table table-bordered table-striped stopped-table"
    >
      <thead>
        <tr>
          <th />
          <th>
            Total
          </th>
          <th>
            24h
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          className="error-row"
          key="error"
        >
          <td>
            Error
          </td>
          <td
            className="total-col active"
            onClick={[Function]}
          >
            0
          </td>
          <td
            className="last-col"
            onClick={[Function]}
          >
            0
          </td>
        </tr>
        <tr
          className="warning-row"
          key="warning"
        >
          <td>
            Warning
          </td>
          <td
            className="total-col"
            onClick={[Function]}
          >
            0
          </td>
          <td
            className="last-col"
            onClick={[Function]}
          >
            0
          </td>
        </tr>
        <tr
          className="success-row"
          key="success"
        >
          <td>
            Success
          </td>
          <td
            className="total-col"
            onClick={[Function]}
          >
            0
          </td>
          <td
            className="last-col"
            onClick={[Function]}
          >
            0
          </td>
        </tr>
      </tbody>
    </table>
  </CardBody>
</Card>
`;

exports[`StoppedTasksCard render selected 1`] = `
<Card
  accented={false}
  aggregated={false}
  aggregatedMini={false}
  cardRef={null}
  className="tasks-donut-card stopped-tasks-card selected-tasks-card"
  matchHeight={false}
>
  <CardTitle
    className=""
    onClick={[Function]}
  >
    Stopped
  </CardTitle>
  <CardBody
    className=""
  >
    <table
      className="table table-bordered table-striped stopped-table"
    >
      <thead>
        <tr>
          <th />
          <th>
            Total
          </th>
          <th>
            24h
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          className="error-row"
          key="error"
        >
          <td>
            Error
          </td>
          <td
            className="total-col"
            onClick={[Function]}
          >
            0
          </td>
          <td
            className="last-col"
            onClick={[Function]}
          >
            0
          </td>
        </tr>
        <tr
          className="warning-row"
          key="warning"
        >
          <td>
            Warning
          </td>
          <td
            className="total-col"
            onClick={[Function]}
          >
            0
          </td>
          <td
            className="last-col"
            onClick={[Function]}
          >
            0
          </td>
        </tr>
        <tr
          className="success-row"
          key="success"
        >
          <td>
            Success
          </td>
          <td
            className="total-col"
            onClick={[Function]}
          >
            0
          </td>
          <td
            className="last-col"
            onClick={[Function]}
          >
            0
          </td>
        </tr>
      </tbody>
    </table>
  </CardBody>
</Card>
`;

exports[`StoppedTasksCard render success-last selected 1`] = `
<Card
  accented={false}
  aggregated={false}
  aggregatedMini={false}
  cardRef={null}
  className="tasks-donut-card stopped-tasks-card selected-tasks-card"
  matchHeight={false}
>
  <CardTitle
    className=""
    onClick={[Function]}
  >
    Stopped
  </CardTitle>
  <CardBody
    className=""
  >
    <table
      className="table table-bordered table-striped stopped-table"
    >
      <thead>
        <tr>
          <th />
          <th>
            Total
          </th>
          <th>
            week
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          className="error-row"
          key="error"
        >
          <td>
            Error
          </td>
          <td
            className="total-col"
            onClick={[Function]}
          >
            0
          </td>
          <td
            className="last-col"
            onClick={[Function]}
          >
            0
          </td>
        </tr>
        <tr
          className="warning-row"
          key="warning"
        >
          <td>
            Warning
          </td>
          <td
            className="total-col"
            onClick={[Function]}
          >
            0
          </td>
          <td
            className="last-col"
            onClick={[Function]}
          >
            0
          </td>
        </tr>
        <tr
          className="success-row"
          key="success"
        >
          <td>
            Success
          </td>
          <td
            className="total-col"
            onClick={[Function]}
          >
            0
          </td>
          <td
            className="last-col active"
            onClick={[Function]}
          >
            0
          </td>
        </tr>
      </tbody>
    </table>
  </CardBody>
</Card>
`;

exports[`StoppedTasksCard render success-total selected 1`] = `
<Card
  accented={false}
  aggregated={false}
  aggregatedMini={false}
  cardRef={null}
  className="tasks-donut-card stopped-tasks-card selected-tasks-card"
  matchHeight={false}
>
  <CardTitle
    className=""
    onClick={[Function]}
  >
    Stopped
  </CardTitle>
  <CardBody
    className=""
  >
    <table
      className="table table-bordered table-striped stopped-table"
    >
      <thead>
        <tr>
          <th />
          <th>
            Total
          </th>
          <th>
            24h
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          className="error-row"
          key="error"
        >
          <td>
            Error
          </td>
          <td
            className="total-col"
            onClick={[Function]}
          >
            0
          </td>
          <td
            className="last-col"
            onClick={[Function]}
          >
            0
          </td>
        </tr>
        <tr
          className="warning-row"
          key="warning"
        >
          <td>
            Warning
          </td>
          <td
            className="total-col"
            onClick={[Function]}
          >
            0
          </td>
          <td
            className="last-col"
            onClick={[Function]}
          >
            0
          </td>
        </tr>
        <tr
          className="success-row"
          key="success"
        >
          <td>
            Success
          </td>
          <td
            className="total-col active"
            onClick={[Function]}
          >
            0
          </td>
          <td
            className="last-col"
            onClick={[Function]}
          >
            0
          </td>
        </tr>
      </tbody>
    </table>
  </CardBody>
</Card>
`;

exports[`StoppedTasksCard render warning-last selected 1`] = `
<Card
  accented={false}
  aggregated={false}
  aggregatedMini={false}
  cardRef={null}
  className="tasks-donut-card stopped-tasks-card selected-tasks-card"
  matchHeight={false}
>
  <CardTitle
    className=""
    onClick={[Function]}
  >
    Stopped
  </CardTitle>
  <CardBody
    className=""
  >
    <table
      className="table table-bordered table-striped stopped-table"
    >
      <thead>
        <tr>
          <th />
          <th>
            Total
          </th>
          <th>
            week
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          className="error-row"
          key="error"
        >
          <td>
            Error
          </td>
          <td
            className="total-col"
            onClick={[Function]}
          >
            0
          </td>
          <td
            className="last-col"
            onClick={[Function]}
          >
            0
          </td>
        </tr>
        <tr
          className="warning-row"
          key="warning"
        >
          <td>
            Warning
          </td>
          <td
            className="total-col"
            onClick={[Function]}
          >
            0
          </td>
          <td
            className="last-col active"
            onClick={[Function]}
          >
            0
          </td>
        </tr>
        <tr
          className="success-row"
          key="success"
        >
          <td>
            Success
          </td>
          <td
            className="total-col"
            onClick={[Function]}
          >
            0
          </td>
          <td
            className="last-col"
            onClick={[Function]}
          >
            0
          </td>
        </tr>
      </tbody>
    </table>
  </CardBody>
</Card>
`;

exports[`StoppedTasksCard render warning-total selected 1`] = `
<Card
  accented={false}
  aggregated={false}
  aggregatedMini={false}
  cardRef={null}
  className="tasks-donut-card stopped-tasks-card selected-tasks-card"
  matchHeight={false}
>
  <CardTitle
    className=""
    onClick={[Function]}
  >
    Stopped
  </CardTitle>
  <CardBody
    className=""
  >
    <table
      className="table table-bordered table-striped stopped-table"
    >
      <thead>
        <tr>
          <th />
          <th>
            Total
          </th>
          <th>
            24h
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          className="error-row"
          key="error"
        >
          <td>
            Error
          </td>
          <td
            className="total-col"
            onClick={[Function]}
          >
            0
          </td>
          <td
            className="last-col"
            onClick={[Function]}
          >
            0
          </td>
        </tr>
        <tr
          className="warning-row"
          key="warning"
        >
          <td>
            Warning
          </td>
          <td
            className="total-col active"
            onClick={[Function]}
          >
            0
          </td>
          <td
            className="last-col"
            onClick={[Function]}
          >
            0
          </td>
        </tr>
        <tr
          className="success-row"
          key="success"
        >
          <td>
            Success
          </td>
          <td
            className="total-col"
            onClick={[Function]}
          >
            0
          </td>
          <td
            className="last-col"
            onClick={[Function]}
          >
            0
          </td>
        </tr>
      </tbody>
    </table>
  </CardBody>
</Card>
`;

exports[`StoppedTasksCard render with minimal props 1`] = `
<Card
  accented={false}
  aggregated={false}
  aggregatedMini={false}
  cardRef={null}
  className="tasks-donut-card stopped-tasks-card"
  matchHeight={false}
>
  <CardTitle
    className=""
    onClick={[Function]}
  >
    Stopped
  </CardTitle>
  <CardBody
    className=""
  >
    <table
      className="table table-bordered table-striped stopped-table"
    >
      <thead>
        <tr>
          <th />
          <th>
            Total
          </th>
          <th>
            24h
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          className="error-row"
          key="error"
        >
          <td>
            Error
          </td>
          <td
            className="total-col"
            onClick={[Function]}
          >
            0
          </td>
          <td
            className="last-col"
            onClick={[Function]}
          >
            0
          </td>
        </tr>
        <tr
          className="warning-row"
          key="warning"
        >
          <td>
            Warning
          </td>
          <td
            className="total-col"
            onClick={[Function]}
          >
            0
          </td>
          <td
            className="last-col"
            onClick={[Function]}
          >
            0
          </td>
        </tr>
        <tr
          className="success-row"
          key="success"
        >
          <td>
            Success
          </td>
          <td
            className="total-col"
            onClick={[Function]}
          >
            0
          </td>
          <td
            className="last-col"
            onClick={[Function]}
          >
            0
          </td>
        </tr>
      </tbody>
    </table>
  </CardBody>
</Card>
`;

exports[`StoppedTasksCard render with props 1`] = `
<Card
  accented={false}
  aggregated={false}
  aggregatedMini={false}
  cardRef={null}
  className="tasks-donut-card stopped-tasks-card"
  matchHeight={false}
>
  <CardTitle
    className=""
    onClick={[Function]}
  >
    Stopped
  </CardTitle>
  <CardBody
    className=""
  >
    <table
      className="table table-bordered table-striped stopped-table"
    >
      <thead>
        <tr>
          <th />
          <th>
            Total
          </th>
          <th>
            week
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          className="error-row"
          key="error"
        >
          <td>
            Error
          </td>
          <td
            className="total-col"
            onClick={[Function]}
          >
            9
          </td>
          <td
            className="last-col"
            onClick={[Function]}
          >
            1
          </td>
        </tr>
        <tr
          className="warning-row"
          key="warning"
        >
          <td>
            Warning
          </td>
          <td
            className="total-col"
            onClick={[Function]}
          >
            8
          </td>
          <td
            className="last-col"
            onClick={[Function]}
          >
            2
          </td>
        </tr>
        <tr
          className="success-row"
          key="success"
        >
          <td>
            Success
          </td>
          <td
            className="total-col"
            onClick={[Function]}
          >
            7
          </td>
          <td
            className="last-col"
            onClick={[Function]}
          >
            3
          </td>
        </tr>
      </tbody>
    </table>
  </CardBody>
</Card>
`;
